src/semantic_world/world_description/world_entity.py

```
@@ -197,19 +175,20 @@ class Body(KinematicStructureEntity, SubclassJSONSerializer):
     The index of the entity in `_world.kinematic_structure`.
     """
 
-    @cached_property
-    def combined_collision_mesh(self) -> Trimesh:
-        """
-        Combines all collision meshes into a single mesh, applying the respective transformations.
-        :return: A single Trimesh representing the combined collision geometry.
-        """
-        transformed_meshes = []
-        for shape in self.collision:
-            transform = shape.origin.to_np()
-            mesh = shape.mesh.copy()
-            mesh.apply_transform(transform)
-            transformed_meshes.append(mesh)
-        return concatenate(transformed_meshes)
+    def __post_init__(self):
+        if not self.name:
+            self.name = PrefixedName(f"body_{id_generator(self)}")
+
+        if self._world is not None:
+            self.index = self._world.kinematic_structure.add_node(self)
+
+        for c in self.collision:
+            c.origin.reference_frame = self
```

same issue here as in shapecollection. just overriding the reference frame without checking can cause really frustrating and hard to debug errors. maybe throw an error if the reference frame is set and is not self, or transform the shape into the correct frame